### ISSUE TITLE
[Feature request] Verbose per-method argument tracing

### ISSUE THREAD

**Issue by nsivov** [doitsujin/dxvk#869]
I'd be interested in extensive tracing capability, when every method call is logged, similar to Wine +d3d,+d3d11 channels. Was that ever considered?

**Comment by doitsujin** [comment]
> Was that ever considered?

Yes, and in the very early days there was a way to do it, but it was removed and never reintroduced for a number of reasons.

**a)** It's a **lot** of work to manually instrument every single method, and clobbers the code with non-essential stuff.
**b)** For debugging purposes, *only* instrumenting the D3D11 and DXGI entry points is next to useless since the actual work is being off-loaded to a separate thread. Which means that a lot of the backend methods would have to be logged as well.
**c)** With b) in mind, it will have a significant considerable impact even when logging is disabled. Making it debug build only or something like that only contributes to the problems metioned in a).
**d)** Apitrace captures everything that's relevant.
